packages/near-membrane-base/src/intrinsics.ts

```diff
@@ -158,3 +160,13 @@ export function getFilteredEndowmentDescriptors(endowments: object): PropertyDes
     }
     return to;
 }
+
+export function getResolvedShapeDescriptors(
+    globalObjectShape: typeof globalThis,
+    endowments: object
+): PropertyDescriptorMap {
+    const globalObjectShapeDescriptors = getFilteredGlobalObjectShapeDescriptors(globalObjectShape);
+    const endowmentsDescriptors = getFilteredGlobalObjectShapeDescriptors(endowments);
+    // eslint-disable-next-line prefer-object-spread
+    return ObjectAssign({ __proto__: null }, globalObjectShapeDescriptors, endowmentsDescriptors);
```

Comment:
the issue that I noticed here is about `ReflectiveIntrinsicObjectNames`, those should never be remapped because they are linked directly. I don't think it is really a problem because they get linked first, but it is unnecessary work since it will be a noop, maybe just adding them to the conditions.

Reply:
Fixed!